### PR TITLE
Refactor remaining lists to hooks

### PR DESCRIPTION
**Pull request by keithjgrant**
##### SUMMARY
Refactor TeamList, ProjectList, and InventoryList to use hooks, applying bug fixes made in `useRequest` and `useDeleteItems`

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
 - UI

### CODE REVIEW

File: awx/ui_next/src/screens/Inventory/InventoryList/InventoryList.test.jsx [hunk 1]
```diff
@@ -140,186 +141,174 @@ describe('<InventoriesList />', () => {
...
 
-  test('error is shown when inventory not successfully deleted from api', async done => {
+  test('should show deletion error', async () => {
     InventoriesAPI.destroy.mockRejectedValue(
       new Error({
         response: {
           config: {
             method: 'delete',
-            url: '/api/v2/inventories/1',
+            url: '/api/v2/teams/1',
```
Comment (mabashian) [inline awx/ui_next/src/screens/Inventory/InventoryList/InventoryList.test.jsx]: I don't think the actual error response matters here but the url would be `/api/v2/inventories/1` - this is probably some copy/paste (?)

File: awx/ui_next/src/screens/Inventory/InventoryList/InventoryList.jsx [hunk 2]
```diff
@@ -23,225 +24,172 @@ const QS_CONFIG = getQSConfig('inventory', {
...
+              showExpandCollapse
+              isAllSelected={isAllSelected}
+              onSelectAll={handleSelectAll}
+              qsConfig={QS_CONFIG}
+              additionalControls={[
+                <ToolbarDeleteButton
+                  key="delete"
+                  onDelete={handleInventoryDelete}
+                  itemsToDelete={selected}
+                  pluralizedItemName="Inventories"
```
Comment (mabashian) [inline awx/ui_next/src/screens/Inventory/InventoryList/InventoryList.jsx]: I know this was like this before you got in here but do you mind marking this one for translation?

**Comment by mabashian** [comment]
@keithjgrant - looks great.  Left a couple of nit-picks but otherwise this lgtm.

**Review by mabashian — APPROVED** [review]

**Comment by mabashian** [comment]
regate

**Comment by mabashian** [comment]
regate

**Comment by jakemcdermott** [comment]
:middle_finger:  regate :middle_finger:

**Comment by jakemcdermott** [comment]
:middle_finger: regate :middle_finger:

**Pull request merged**